File: Ruby/P4/Irrgarten/weapon.rb
#encoding:utf-8

=begin

    Autor: Isabel Morro Tabares
    
    Armas que utiliza el jugador en los ataques durante los combates

=end

require_relative 'dice'
require_relative 'combat_element'

module Irrgarten
    class Weapon < CombatElement

        # Constructor

        def initialize (power, uses)
           super
        end

        # Métodos públicos

        def attack 
            produce_effect
        end

        # To string
        def to_s
            "W[" + super
        end
    end

end

File: Java/P5/P5/irrgarten/irrgarten/src/irrgarten/Weapon.java

package irrgarten;

/**
 * @brief   Armas que utiliza el jugador en los ataques durante los combates
 * @author  Isabel Morro Tabares
 */
public class Weapon extends CombatElement {

    // Constructor
    
    public Weapon (float _power, int _uses){
       super(_power, _uses);
    }
    
    // Métodos públicos
    
    public float attack (){
       return produceEffect();
    }

    // To string
    @Override
    public String toString(){
        String cadena = "W[";
        cadena += super.toString();
        return cadena;
    }
}
